From ba1390a80662ff2ab7bfda978cde7df9a871f6ae Mon Sep 17 00:00:00 2001
From: Changqing Li <changqing.li@windriver.com>
Date: Tue, 24 Jul 2018 15:03:39 +0800
Subject: [PATCH] configure.ac: allow cross-compilation

The checking OpenSSL library and header version consistency will
always fail in cross compiling, skip the check and give a warning
instead for cross compiling.

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
Signed-off-by: Yi Zhao <yi.zhao@windriver.com>

Update to new version 3.0.17 to fix patch warning
Signed-off-by: Changqing Li <changqing.li@windriver.com>

Rebased for 3.2.3
Signed-off-by: Randy MacLeod <randy.macleod@windriver.com>
Signed-off-by: Robert Yang <liezhi.yang@windriver.com>

---
 src/modules/rlm_krb5/configure.ac | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/modules/rlm_krb5/configure.ac b/src/modules/rlm_krb5/configure.ac
index 9ee6379ea4..1dad481da8 100644
--- a/src/modules/rlm_krb5/configure.ac
+++ b/src/modules/rlm_krb5/configure.ac
@@ -143,7 +143,8 @@ if test "$krb5threadsafe" != "no"; then
 	FR_SMART_CHECK_LIB(krb5, krb5_is_thread_safe)
 	if test "x$ac_cv_lib_krb5_krb5_is_thread_safe" = xyes; then
 		AC_RUN_IFELSE([AC_LANG_PROGRAM([[#include <krb5.h>]], [[return krb5_is_thread_safe() ? 0 : 1]])],
-			[krb5threadsafe="-DKRB5_IS_THREAD_SAFE"], [AC_MSG_WARN([[libkrb5 is not threadsafe]])])
+			[krb5threadsafe="-DKRB5_IS_THREAD_SAFE"], [AC_MSG_WARN([[libkrb5 is not threadsafe]])],
+			[AC_MSG_WARN(cross compiling: not checking)])
 	fi
 else
 	krb5threadsafe=""
